src/Services/FHIR/FhirProcedureService.php

```diff
@@ -0,0 +1,209 @@
+                        porder.encounter_id,
+                        porder.date_collected,
+                        porder.provider_id,
+                        ptype.body_site,
+                        ptype.related_code,
+                        ptype.notes,
+                        porder.order_diagnosis,
+                        porder.order_status,
+                        presult.result_status
+                        FROM procedure_order AS porder 
+                        LEFT JOIN procedure_order_code AS pcode 
+                        ON porder.procedure_order_id = pcode.procedure_order_id 
+                        LEFT JOIN procedure_type AS ptype 
+                        ON pcode.procedure_code = ptype.procedure_code 
+                        LEFT JOIN procedure_report AS preport 
+                        ON preport.procedure_order_id = porder.procedure_order_id 
+                        LEFT JOIN procedure_result AS presult 
+                        ON presult.procedure_report_id = preport.procedure_report_id";
+    }
+
+    public function getAll($search)
+    {
+        $SQL = $this->get();
+        if (isset($search['patient'])) {
+            $SQL .= " WHERE porder.patient_id = ?;";
+        }
+
+        $procedureResults = sqlStatement($SQL, $search['patient']);
+        $results = array();
+        while ($row = sqlFetchArray($procedureResults)) {
+            $SQLProvider = "SELECT specialty, organization FROM users WHERE id = ?";
+            $provider = sqlQuery($SQLProvider, array($row['provider_id']));
+            $row['function'] = $provider['specialty'];
+            $row['organization'] = $provider['organization'];
+            array_push($results, $row);
+        }
+
+        return $results;
+    }
+
+    public function getOne($id)
+    {
+        $SQL = $this->get();
+        $SQL .= " WHERE porder.procedure_order_id = ? ";
+
+        $sqlResult = sqlStatement($SQL, $id);
+        $result = sqlFetchArray($sqlResult);
+
+        $SQLProvider = "SELECT specialty, organization FROM users WHERE id = ?";
+        $provider = sqlQuery($SQLProvider, array($result['provider_id']));
+        $result['function'] = $provider['specialty'];
+        $result['organization'] = $provider['organization'];
+        
+        return $result;
+    }
+
+    public function createProcedureResource($id = '', $data = '', $encode = true)
+    {
+        $status = new FHIREventStatus();
+        if ($data['order_status'] == "completed") {
+            $status->setValue("completed");
+        } else if ($data['order_status'] == "pending") {
+            $status->setValue("in-progress");
+        } else if ($data['order_status'] == "cancelled") {
+            $status->setValue("stopped");
+        } else {
+            $status->setvalue("Unknown");
+        }
+
+        $procedureCode = new FHIRCodeableConcept();
+        $procedureCodeCoding = new FHIRCoding();
+        $procedureCodeCoding->setSystem("http://snomed.info/sct");
```

Comment:
Not every code is sct, in this case they are cpt(procedure code). Well almost always.
We rarely use sct mainly because of billing and sct was a baby when most of these sources were written.
Plus providers hate them:)

We may need to do a code type lookup to determine for all our codes. Perhaps a new routine in FHIRResorceService.

Reply:
I'll fix it to cpt. 

> We may need to do a code type lookup to determine for all our codes. Perhaps a new routine in FHIRResorceService.

Yes that would help